cmd/skopeo/sync.go

```diff
@@ -0,0 +1,594 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/copy"
+	"github.com/containers/image/directory"
+	"github.com/containers/image/docker"
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+type syncOptions struct {
```

Comment:
Although not strictly needed for go-docs, I'd love to have comments for all the types and their fields where the use and intention is not obvious. It'll help with future maintenance.